modules/images/webp-uploads/load.php

```diff
@@ -8,6 +8,13 @@
  * @package performance-lab
  */
 
+// Define the constant.
+if ( defined( 'WEBP_UPLOADS_VERSION' ) ) {
+	return;
+}
+
+define( 'WEBP_UPLOADS_VERSION', '1.0.0' );
+
 // Do not load the code if it is already loaded through another means.
 if ( function_exists( 'webp_uploads_create_sources_property' ) ) {
 	return;
```

Comment:
This can be removed now that the constant is already being defined and checked for. No need to check for two things from the module that are already present, one is sufficient. Maybe move the comment here above the `defined( ...` check?

Reply:
@felixarntz is there a risk that we need to account for where someone has a version of a plugin installed previous to when these constants were implemented where there would be a conflict with an already defined function?

Reply:
@joemcgill That's a great point, I hadn't thought through this. These `function_exists` checks in the modules are just being introduced in next week's release 2.1.0, so they're also new. However, since the work here including the constants is only going into a feature branch, this won't ship in the same release, so your concern could indeed be valid.

We won't be releasing any standalone plugins before this feature branch gets merged back into `trunk`. So the standalone plugins will always have the constants and constant checks from the start. The load order is generally that plugins load first, before the PL modules. So "older" versions of Performance Lab ("older" only in the future, e.g. 2.1.0+ after we have started to publish standalone plugins) would have the `function_exists` check in the modules, and since the function would exist from the standalone plugin loaded before, no fatal error would occur. So this general use-case is fine.

I think the only problem could be when you have a module in the Performance Lab active and _then_ activate the corresponding standalone plugin on the site. The activation of a plugin is a special case where it happens after initializing WordPress, so in that scenario the module would be loaded first. That is where your point indeed becomes relevant: If we _only_ have the constant check in the standalone plugin and an "older" version of PL is used (e.g. upcoming 2.1.0) that does not include the constant, it would lead to a fatal error. So for that reason you're right - we should keep the `function_exists()` check, at least for some time in the future maybe until adoption of those older PL versions has gone to an extremely low level (i.e. probably at least several months from now).

Summaryizing: In general we shouldn't need both checks, but because we first ship only the `function_exists` check, there is a chance for fatal errors if we remove them now.

Thanks for reading to me thinking through this :)

Reply:
Thanks for the explanation, we might consider running a quick set of manual checks by activating the standalone plugins with different versions of the PL plugin installed, but really, I don't know why someone would install and activate these plugins if they already have a version of the PL plugin where these are included as a module, so probably an edge case that we can document as part of the plugin repo readme.